src/governance.h

```diff
@@ -147,11 +163,19 @@ class CGovernanceManager
     void UpdatedBlockTip(const CBlockIndex *pindex);
     int64_t GetLastDiffTime() {return nTimeLastDiff;}
     void UpdateLastDiffTime(int64_t nTimeIn) {nTimeLastDiff=nTimeIn;}
+
+    int GetCachedBlockHeight() { return nCachedBlockHeight; }
```

Comment:
Looks like this can be replaced by `return pCurrentBlockIndex->nHeight;` and then `nCachedBlockHeight` and `UpdateCachedBlockHeight()` can be removed completely. This should probably help to avoid deadlocks too.